File: Data Structures & Algorithms/meeting-schedule/submission-0.cs
/**
 * Definition of Interval:
 * public class Interval {
 *     public int start, end;
 *     public Interval(int start, int end) {
 *         this.start = start;
 *         this.end = end;
 *     }
 * }
 */

public class Solution {
    public bool CanAttendMeetings(List<Interval> intervals) {
        if (intervals.Count == 0)
        {
            return true;
        }

        List<Interval> sorted = intervals.OrderBy(i => i.start).ToList();

        Interval lastInterval = sorted.First();
        foreach (Interval interval in sorted.Skip(1))
        {
            if (interval.start < lastInterval.end)
            {
                return false;
            }

            lastInterval = interval;
        }

        return true;
    }
}   

File: Data Structures & Algorithms/meeting-schedule-ii/submission-0.py
"""
Definition of Interval:
class Interval(object):
    def __init__(self, start, end):
        self.start = start
        self.end = end
"""

class Solution:
    def minMeetingRooms(self, intervals: List[Interval]) -> int:
        # Sort the intervals by start time
        # If the first intervals overlaps the second, the first interval should be it's own "room"
        # Continue iterating with the same idea

        if len(intervals) == 0:
            return 0

        res = 1
        lastInterval = intervals[0]
        for interval in intervals[1:]:
            print(f"{interval.start} {lastInterval.end}")
            if interval.start < lastInterval.end:
                res += 1
            
            lastInterval = interval

        return res
        